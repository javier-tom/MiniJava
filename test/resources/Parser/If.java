class Main {
    public static void main(String[] args) {
        System.out.println(new If().If(5));
    }
}

class If {
    public int If(int max) {
        int i;
        i = 0;
        if (i < max) {
            System.out.println(i);
        } else {
            System.out.println(max);
        }
        return 0;
    }
}