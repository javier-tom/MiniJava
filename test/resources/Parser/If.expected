Program
  MainClass Main (line 1)
    Print (line 3)
      Call If (line 3)
        on class:
          NewObject If (line 3)
        parameters:
          5 (line 3)

  Class If (line 7)
    fields:
    MethodDecl If (line 8)
      returns int
      params:
        int max
      variables:
        VarDecl int i (line 9)
      Assign i (line 10)
        new value:
          0 (line 10)
      If (line 11)
        condition:
          LessThan (line 11)
            i (line 11)
            max (line 11)
        body:
          Print (line 12)
            i (line 12)
        else
          Print (line 14)
            max (line 14)
      Return
        0 (line 16)